pydantic/json_schema.py

```diff
@@ -2042,10 +2042,16 @@ def handle_ref_overrides(self, json_schema: JsonSchemaValue) -> JsonSchemaValue:
         return json_schema
 
     def get_schema_from_definitions(self, json_ref: JsonRef) -> JsonSchemaValue | None:
-        def_ref = self.json_to_defs_refs[json_ref]
-        if def_ref in self._core_defs_invalid_for_json_schema:
-            raise self._core_defs_invalid_for_json_schema[def_ref]
-        return self.definitions.get(def_ref, None)
+        try:
+            def_ref = self.json_to_defs_refs[json_ref]
+            if def_ref in self._core_defs_invalid_for_json_schema:
+                raise self._core_defs_invalid_for_json_schema[def_ref]
+            return self.definitions.get(def_ref, None)
+        except KeyError as e:
+            if json_ref.startswith(('http://', 'https://')):
+                return None
+            else:
+                raise e
```

Comment:
Would recommend this just be `raise` as that's the builtin syntax for reraising an existing exception with its stack intact.

Same for the two below instances as well.